src/main/java/duke/directprocessor/TaskList.java

```diff
@@ -0,0 +1,118 @@
+package duke.directprocessor;
+
+import duke.tasks.Task;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+
+/**
+ * This class holds the current task list input by the user. It is also in charge of save the task list into a file
+ *     when the exit command is given by the user.
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList;
+
+    /**
+     * The constructor of the class to generate an empty task list.
+     * Note that this constructor will be applied only when it fails to read from the file.
+     */
+    public TaskList() {
+        taskList = new ArrayList<>();
```

Comment:
I think it is preferred to include the generic type, i.e. `new ArrayList<Task>()`.